app/actors/hyrax/actors/asset_actor.rb

```diff
@@ -63,6 +63,7 @@ def set_admin_data_attributes(admin_data, env)
 
         def should_empty_admin_data_value?(key, admin_data, env)
           key != :bulkrax_importer_id &&
+          env.attributes.key?(key) &&
```

Comment:
This is a bit confusing.  And thinking about the how we can short circuit calling admin data; I suspect that's slower.

```ruby
key != :bulkrax_importer_id &&
env.attributes.key?(key) &&
!env.attributes[key].present? &&
admin_data.send(key).present?
```

Reply:
> And thinking about the how we can short circuit calling admin data; I suspect that's slower.

@jeremyf can you explain what you mean here? What is slower?

Reply:
Sure.  `env.attributes.key?` is almost certainly a hash access.  Whereas `admin_data` is an object and we're sending it the `key` message.  What does that method do?  It could query all of Fedora.

Also, by having that `admin_data` sandwiched between two `env.attributes` methods there's a bit of a disjointed logical flow; namely, I'm going from one subject to another and then back to that first subject.  And with the `&&` that feels a bit much.